### PR TITLE
#25 Added memory cache mechanism

### PR DESCRIPTION
**Pull request by barissaglam**

### CODE REVIEW

**Comment by barissaglam** [comment]
I will update unit tests

File: app/src/main/java/com/mertceyhan/bitcoinmarket/features/market/di/AppModule.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    ): MarketDatabase =
+        Room.databaseBuilder(context, MarketDatabase::class.java, MarketDatabase.DB_NAME)
+            .build()
+
+    @Singleton
+    @Provides
+    fun provideSharedPreferences(
+        @ApplicationContext context: Context
+    ): SharedPreferences = context.getSharedPreferences(
+        "asd", Context.MODE_PRIVATE
```
Comment (mertceyhan) [inline app/src/main/java/com/mertceyhan/bitcoinmarket/features/market/di/AppModule.kt:31]: Could you change the naming of `SharedPreferences`? The name `asd` doesn't make any sense to me.

**Review by mertceyhan — CHANGES_REQUESTED** [review]

**Comment by barissaglam** [comment]
@mertceyhan Could you  please review again?

**Comment by mertceyhan** [comment]
Could you check `MarketRepositoryImpTest`, please? @barissaglam

**Comment by barissaglam** [comment]
> Could you check `MarketRepositoryImpTest`, please? @barissaglam

Could you check again?

**Review by mertceyhan — APPROVED** [review]

**Pull request merged**